aptos-move/aptos-vm/src/adapter_common.rs

```diff
@@ -51,7 +52,21 @@ pub(crate) trait VMAdapter {
     fn execute_single_transaction(
         &self,
         txn: &PreprocessedTransaction,
-        data_cache: &impl MoveResolverExt,
+        data_cache: &impl StateView,
+        log_context: &AdapterLogSchema,
+        aggregator_enabled: bool,
+    ) -> Result<(VMStatus, VMOutput, Option<String>), VMStatus>;
+
+    /// Execute a single transaction when running the block sequentially.
+    /// This function will first run the `execute_single_transaction` function
+    /// with aggregators enabled. If there is an aggregator error (overflow/underflow),
+    /// we are not sure if the error is triggered in the right location in the code.
+    /// Therefore, for proper error handlig, we will re-run the `execute_transaction`
+    /// function with aggregators disabled and output the final output.
+    fn execute_single_transaction_sequential(
```

Comment:
I don't like _sequential and _parallel in the names of the functions, and needing to read comments to understand what they do.

It would be better if function explains what it does, instead of who calls it.

maybe:
execute_single_transaction => execute_single_transaction_one_pass
execute_single_transaction_sequential => execute_single_transaction_with_materialization

or any better names that you can think of :)

Reply:
Based on Rati's comments, we will move this logic to executor.rs.